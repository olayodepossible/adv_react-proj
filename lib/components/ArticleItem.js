import React from "react";

const ArticleItem = ({ article, author }) => {
  return (
    <div>
      <h3>{article.title}</h3>
      <h4>{article.date}</h4>
      <div>
        <a href={author.website}>
          <h4>
            {author.firstName} - {author.lastName}
          </h4>
        </a>
      </div>
      <p>{article.body}</p>
    </div>
  );
};

export default ArticleItem;
