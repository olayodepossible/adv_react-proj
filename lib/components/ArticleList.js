import React from "react";
import ArticleItem from "./ArticleItem";

const ArticleList = ({ articles, authors }) => {
  return (
    <div>
      {Object.values(articles).map((article) => (
        <ArticleItem article={article} author={authors[article.authorId]} />
      ))}
    </div>
  );
};

export default ArticleList;
