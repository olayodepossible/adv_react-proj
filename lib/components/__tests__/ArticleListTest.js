import React from "react";
import ArticleItem from "../Article";
import renderer from "react-test-renderer";
import ArticleList from "../ArticleList";
import { expectedError } from "@babel/core/lib/errors/rewrite-stack-trace";

describe("ArticleList", () => {
  const testProps = {
    articles: {
      a: { id: "a" },
      b: { id: "b" },
    },
    articleActions: {
      getAuthor: jest.fn().mockImplementation(() => ({
        id: "d85577ea34ae50f2dac5347b5219aa23",
        firstName: "Andrew",
        lastName: "Clark",
        website: "https://twitter.com/acdlite",
      })),
    },
  };

  it("renders correctly", () => {
    const tree = renderer.create(<ArticleList {...testProps} />).toJSON();
    expect(tree.children.length).toBe(2);
    expect(tree).toMatchSnapshot();
  });
});
