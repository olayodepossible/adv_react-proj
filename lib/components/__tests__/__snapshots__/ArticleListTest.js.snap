// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div
  className=""
>
  <div
    style={
      {
        "borderBottomColor": " #aaa",
        "borderBottomSize": 1,
        "borderBottomStyle": "solid",
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <h3
      style={
        {
          "fontWeight": "bold",
        }
      }
    />
    <h4
      style={
        {
          "colo": "#888",
          "fontSize": "0.85rem",
        }
      }
    >
      Invalid Date
    </h4>
    <div
      style={
        {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href="https://twitter.com/acdlite"
      >
        <h4>
          Andrew
           - 
          Clark
        </h4>
      </a>
    </div>
    <p
      style={
        {
          "paddingLeft": 20,
        }
      }
    />
  </div>
  <div
    style={
      {
        "borderBottomColor": " #aaa",
        "borderBottomSize": 1,
        "borderBottomStyle": "solid",
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <h3
      style={
        {
          "fontWeight": "bold",
        }
      }
    />
    <h4
      style={
        {
          "colo": "#888",
          "fontSize": "0.85rem",
        }
      }
    >
      Invalid Date
    </h4>
    <div
      style={
        {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href="https://twitter.com/acdlite"
      >
        <h4>
          Andrew
           - 
          Clark
        </h4>
      </a>
    </div>
    <p
      style={
        {
          "paddingLeft": 20,
        }
      }
    />
  </div>
</div>
`;
